docs/healthcheck.md

```diff
@@ -0,0 +1,171 @@
+
+### `/_/healthcheck`
+
+Basic healthchecks return details on the health of Kafka and its topics.
+
+If no HTTP response error, the structure of the response is an array of 3
+key:value objects.
+
+One of these key:value objects will have keys of numbers, and values with an
+object of keys with `nodeId`, `host`, and `port`. The number keys represent
+each zookeeper node, and their respective details.
+
+```
+zookeeperNode: {
+    nodeId: <value>,
+    host: <value>,
+    port: <value>
+}
+```
+
+One of these key:value objects will have a key called `metadata`. The value
+is an object where each key represents each zookeeper node, and each value holds
+details on topic name, partition number, leader number, replica count, and
+in-sync replicas per partition.
+
+```
+metadata: {
+    zookeeperNode: {
+        topic: <value>,
+        partition: <value>,
+        leader: <value>
+    }
+}
+```
+
+One of these key:value objects will have a key called `internalConnections`.
+The value is an object with 3 keys: `isrHealth` which will be either "ok" or
+"error", `zookeeper` which shows a status and details on the status (more about
+zookeeper status codes
+[here at node-zookeeper-client#state](https://github.com/alexguan/node-zookeeper-client#state)),
+and `kafkaProducer` which will be either "ok" or "error" and checks the health
+of all Producers for every topic.
+
+```
+internalConnections: {
+    isrHealth: <ok || error>,
+    zookeeper: {
+        status: <ok || error>,
+        details: {
+            name: <value>,
+            code: <value>
+        }
+    },
+    kafkaProducer: {
+        status: <ok || error>
+    }
+}
+```
+
+**Example Output**:
+(NOTE: some sections in the example below are contracted to reduce redundancy)
+
+```
+[
+    {
+        "0":{
+            "nodeId":0,
+            "host":"server-node1",
+            "port":9092
+        },
+        ...
+        "4":{
+            "nodeId":4,
+            "host":"server-node5",
+            "port":9092
+        }
+    },
+    {
+        "metadata": {
+            "0":{
+                "topic":"backbeat-replication",
+                "partition":0,
+                "leader":4,
+                "replicas":[0,1,4],
+                "isr":[1,4,0]
+            },
+            ...
+            "4":{
+                "topic":"backbeat-replication",
+                "partition":4,
+                "leader":3,
+                "replicas":[0,3,4],
+                "isr":[4,3,0]
+            }
+        }
+    },
+    {
+        "internalConnections":{
+            "isrHealth":"ok",
+            "zookeeper":{
+                "status":"ok",
+                "details":{
+                    "name":"SYNC_CONNECTED",
+                    "code":3
+                }
+            },
+            "kafkaProducer":{
+                "status":"ok"
+            }
+        }
+    }
+]
+```
+
+### `/_/healthcheck/deep`
+
+Deep healthchecks will return the health of every available partition for the
```

Comment:
Same, use the present tense (although deep healthcheck is not yet in the release branch, it will be eventually).